SignalShareExtension/ShareViewController.swift

```diff
@@ -719,6 +720,15 @@ public class ShareViewController: UIViewController, ShareViewDelegate, SAEFailed
                 if ShareViewController.itemMatchesSpecificUtiType(itemProvider: itemProvider,
                                                                   utiType: kUTTypeVCard as String) {
                     customFileName = "Contact.vcf"
+
+                    if let contactShare = OWSContacts.contact(forVCardData: data) {
+                        isConvertibleToContactShare = true
+                    } else {
+                        Logger.error("\(strongSelf.logTag) could not parse vcard.")
+                        let writeError = ShareViewControllerError.assertionError(description: "Could not parse vcard data.")
+                        reject(writeError)
```

Comment:
The contact-conversion logic is pretty permissive. This should fail only for corrupt vcards or vcards that contain no contacts.  If it contains even a single contact with no valid fields (if that's even possible), it should succeed - and the user will be blocked from sending later with an explicit error message that at least one phone number, email or address should be defined.